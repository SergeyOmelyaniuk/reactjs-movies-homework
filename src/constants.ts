import { Language } from './types';

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const translate: any = {
	en: {
		title: 'title',
		placeholder: 'Movies, person, movie theaters',
		categories: [
			{ id: 1, value: 'popular', title: 'Popular' },
			{ id: 2, value: 'top_rated', title: 'Top rated' },
			{ id: 3, value: 'upcoming', title: 'Upcoming' },
		],
	},
	ru: {
		title: 'заголовок',
		placeholder: 'Фильмы, люди, кинотеатры',
		categories: [
			{ id: 1, value: 'popular', title: 'Популярные' },
			{ id: 2, value: 'top_rated', title: 'Топ рейтинга' },
			{ id: 3, value: 'upcoming', title: 'Ближайшие' },
		],
	},
};
