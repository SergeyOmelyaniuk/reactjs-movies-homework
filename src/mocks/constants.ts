import { Category, Genre, Language } from '../types';

export const categories: Category[] = [
	{ id: 1, value: 'popular', title: 'Popular' },
	{ id: 2, value: 'top_rated', title: 'Top rated' },
	{ id: 3, value: 'upcoming', title: 'Upcoming' },
];

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const genres: Genre[] = [
	{ id: 1, name: 'Genre 1' },
	{ id: 2, name: 'Genre 2' },
	{ id: 3, name: 'Genre 3' },
	{ id: 4, name: 'Genre 4' },
	{ id: 5, name: 'Genre 5' },
];
