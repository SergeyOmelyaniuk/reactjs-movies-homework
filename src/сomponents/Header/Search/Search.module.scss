// .search {}
.input {
	width: 730px;
	height: 50px;
	color: rgb(32, 32, 32);
	padding: 15px;
	background: rgb(255, 255, 255);
	border-radius: 5px 5px 5px 5px;
	font-size: 16px;
}
