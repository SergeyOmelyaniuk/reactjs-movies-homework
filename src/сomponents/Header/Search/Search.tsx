import React, { FormEvent, useState } from 'react';
import styles from './Search.module.scss';

interface SearchProps {
	searchValue: string;
	setSearcValue: (value: string) => void;
}

const Search = ({ searchValue, setSearcValue }: SearchProps) => {
	const [searchInput, setSearchInput] = useState(searchValue);

	const onSubmit = (event: FormEvent) => {
		event.preventDefault();
		setSearcValue(searchInput);
	};

	const onChange = (event: FormEvent) => {
		const { value } = event.target as HTMLInputElement;
		setSearchInput(value);
	};

	return (
		<form className={styles.search} onSubmit={onSubmit}>
			<input
				id=''
				className={styles.input}
				type='text'
				placeholder='Movies, person, movie theaters'
				value={searchInput}
				onChange={onChange}
			/>
			<button type='submit' className={styles.button}></button>
		</form>
	);
};

export default Search;
