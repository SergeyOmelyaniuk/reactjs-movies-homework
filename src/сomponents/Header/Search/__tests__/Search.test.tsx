import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { translate } from '../../../../constants';
import Search from '..';

jest.mock('react-router-dom', () => ({
	useHistory: () => ({
		push: jest.fn(),
	}),
}));

describe('Search', () => {
	const setSearchValueMock = jest.fn();

	test('should render', () => {
		const { container } = render(
			<Search
				placeholder={translate.en.placeholder}
				searchValue=''
				setSearcValue={setSearchValueMock}
			/>
		);

		expect(container).toMatchSnapshot();
	});

	test('should call change value for search on submit event', () => {
		const { getByPlaceholderText, getByRole } = render(
			<Search
				placeholder={translate.en.placeholder}
				searchValue=''
				setSearcValue={setSearchValueMock}
			/>
		);

		const inputElement = getByPlaceholderText('Movies, person, movie theaters');
		fireEvent.change(inputElement, { target: { value: 'Hello' } });
		const buttonElement = getByRole('button');
		fireEvent.click(buttonElement);

		expect(setSearchValueMock).toHaveBeenCalledTimes(1);
		expect(setSearchValueMock).toHaveBeenCalledWith('Hello');
	});
});
