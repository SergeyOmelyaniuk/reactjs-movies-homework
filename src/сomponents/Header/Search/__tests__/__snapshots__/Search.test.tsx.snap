// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render 1`] = `
<div>
  <form
    class="search"
  >
    <input
      class="input"
      placeholder="Movies, person, movie theaters"
      type="text"
      value=""
    />
    <button
      class="button"
      type="submit"
    />
  </form>
</div>
`;
